### PR TITLE
chore: Update `@metamask/gas-fee-controller` and peer deps

### PR DESCRIPTION
**Pull request by Gudahtt**
## **Description**

Update the `@metamask/gas-fee-controller` package to v21 to satisfy the peer dependency on `@metamask/network-controller@^21.0.0`, and update the `@metamask/user-operation-controller` to satisfy its peer dependency upon `@metamask/gas-fee-controller`.

Note that an older version of `@metamask/gas-fee-controller` (v18) remains in the dependency tree, but only because it's imported by `@metamask/smart-transaction-controller` for type reasons. It has no runtime impact on the application, so the associated peer dependency warnings from this older release can be ignored. This will be eliminated soon, in an upcoming PR.

The updated `@metamask/user-operation-controller` still does not have its peer dependencies satisfied, but the problems are pre-existing. The `@metamask/keyring-controller` and `@metamask/transaction-controller` packages are head of where this package expects them to be. This is not made worse by this PR though, and will be addressed in a future PR.

Changelogs:
- `@metamask/gas-fee-controller@21`: https://github.com/MetaMask/core/blob/main/packages/gas-fee-controller/CHANGELOG.md#2100
  - One breaking change with an impact:
    >The inherited AbstractPollingController method startPollingByNetworkClientId has been renamed to startPolling (https://github.com/MetaMask/core/pull/4752)
- `@metamask/user-operation-controller@16`: https://github.com/MetaMask/core/blob/main/packages/user-operation-controller/CHANGELOG.md#1600
  - No breaking changes with an impact. It appeared at first that the `startPollingByNetworkClientId` was breaking, but the user operation controller had an override for that method so it was preserved ([see here](https://github.com/MetaMask/core/pull/4752/files#diff-335af05ece636eb593b348e369dff139dfbfea49ad4e9ba3bb47b4909aab9aefR304))

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28745?quickstart=1)

## **Related issues**

Related to https://github.com/MetaMask/MetaMask-planning/issues/3568

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -241,7 +241,6 @@
     "@expo/config-plugins/glob": "^10.3.10",
     "@solana/web3.js/rpc-websockets": "^8.0.1",
     "@metamask/message-manager": "^10.1.0",
-    "@metamask/gas-fee-controller@npm:^15.1.1": "patch:@metamask/gas-fee-controller@npm%3A15.1.2#~/.yarn/patches/@metamask-gas-fee-controller-npm-15.1.2-db4d2976aa.patch",
```
Comment (Gudahtt) [inline package.json:244]: This resolution was already unused. Not directly related to the bump here, but couldn't hurt to remove it.

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

**Review by mcmire — APPROVED** [review]
Makes sense, looks good.

**Review by micaelae — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]
Looks good 🚀

**Pull request merged**